lib/views/main_screen.dart

```diff
@@ -1,52 +1,277 @@
-// ignore_for_file: talawa_api_doc
+// ignore_for_file: talawa_api_doc, avoid_dynamic_calls
 // ignore_for_file: talawa_good_doc_comments
-
+import 'dart:async';
 import 'package:flutter/material.dart';
+import 'package:hive/hive.dart';
 import 'package:talawa/models/mainscreen_navigation_args.dart';
+import 'package:talawa/plugins/fetch_plugin_list.dart';
+import 'package:talawa/utils/app_localization.dart';
 import 'package:talawa/view_model/main_screen_view_model.dart';
+import 'package:talawa/views/after_auth_screens/add_post_page.dart';
+import 'package:talawa/views/after_auth_screens/chat/chat_list_screen.dart';
+import 'package:talawa/views/after_auth_screens/events/explore_events.dart';
+import 'package:talawa/views/after_auth_screens/feed/organization_feed.dart';
+import 'package:talawa/views/after_auth_screens/profile/profile_page.dart';
 import 'package:talawa/views/base_view.dart';
 import 'package:talawa/widgets/custom_drawer.dart';
 
+/// list of navbarClasses
+late List<StatelessWidget> navBarClasses;
+
+/// list of navBarItems contains actual UI code of navbar items
+late List<BottomNavigationBarItem> navBarItems;
+
 class MainScreen extends StatefulWidget {
   const MainScreen({Key? key, required this.mainScreenArgs}) : super(key: key);
+  //final bool fromSignUp;
   final MainScreenArgs mainScreenArgs;
 
   @override
   State<MainScreen> createState() => _MainScreenState();
 }
 
 class _MainScreenState extends State<MainScreen> {
+  List<List<Object>> renderBottomNavBarPlugins(
+    List<BottomNavigationBarItem> navBarItems,
+    List<StatelessWidget> navBarClasses,
+    Map navNameClasses,
+    Map<String, Widget> navNameIcons,
+    BuildContext context,
+  ) {
+    //get list from hive
+    final Box box;
+    // ignore: prefer_typing_uninitialized_variables
+    var pluginList;
+    const bool res = false;
+    box = Hive.box('pluginBox');
+    pluginList = box.get('plugins');
+
+    /// CAUTION : `pluginList` is currently filled with dummy data in order to make sure that
+    /// all features of talawa should be visible from the navbar.
+    /// following dummy data should be only used for developement purpose.
+    /// Adding dummy data enables to use Events,Posts,Chats features event if plugin documents for them is not created by the admin.
+    /// FOR PRODUCTION : please make this array empty to enable or test real plugin data from the server
+    pluginList ??= [
```

Comment:
Having demo data in the widget class is not a good idea. Having it in a separate `data` file will make the code better.

Reply:
The main focus behind the PR was to bring back the code which was previously deleted.